kubernetes/controller/internal/resolution/workerpool/source.go

```diff
@@ -0,0 +1,107 @@
+package workerpool
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// EventSource is a controller-runtime source that watches the worker pool's event channel
+// and triggers reconciliation for all requesters when a resolution completes.
+type EventSource struct {
+	workerPool *WorkerPool
+}
+
+var _ source.Source = &EventSource{}
+
+// NewEventSource creates a new event source from a worker pool.
+func NewEventSource(workerPool *WorkerPool) *EventSource {
+	return &EventSource{
+		workerPool: workerPool,
+	}
+}
+
+// Start implements source.Source. It starts watching the event channel and enqueues
+// reconciliation requests for all requesters when resolutions complete.
+func (es *EventSource) Start(ctx context.Context, queue workqueue.TypedRateLimitingInterface[reconcile.Request]) error {
```

Comment:
start already runs in a goroutine, so why start another one here? Start is expected to block btw.